WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginActivity.java

```diff
@@ -321,14 +322,26 @@ public void gotXmlRpcEndpoint(String inputSiteAddress, String endpointAddress) {
     }
 
     private void launchHelpshift(String url, String username, boolean isWpcom) {
-        HelpshiftHelper.Tag origin = HelpshiftHelper.chooseHelpshiftLoginTag(getLoginMode() == LoginMode.JETPACK_STATS,
-                isWpcom);
+        HelpshiftHelper.Tag origin;
+        Fragment currentFragment = getSupportFragmentManager().findFragmentById(R.id.fragment_container);
+
+        if (currentFragment instanceof LoginHelpshiftOriginProvider) {
+            origin = ((LoginHelpshiftOriginProvider) currentFragment).helpshiftOriginTag();
+        } else {
+            // For backward compatibility with the old Login activity
```

Comment:
Not sure how the old login activity would interact with `LoginActivity`. The only "old" fragment used here is `NewUserFragment` but it manages launching help on its own. In other words, `launchHelpshift()` be called and `currentFragment` not being a LoginHelpshiftOriginProvider is an invalid state and cannot even happen.